### ISSUE TITLE
Detect the C&C compatibility declaration from third-party plugins

### ISSUE THREAD

**Issue by tarhi-saad** [woocommerce/woocommerce-gutenberg-products-block#8729]
### Context

A couple of months ago, we did a research spike (pdFofs-mB-p2) about managing extensions that don't support the Cart & Checkout Blocks. One of our propositions was to declare (in)compatibility with the C&C Blocks (pdFofs-mB-p2#declare-in-compatibility-with-the-cart-checkout-blocks) using the same approach in [HPOS compatibility declaration](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#declaring-extension-incompatibility). We couldn't use this feature for the Cart & Checkout Block until [this PR](https://github.com/woocommerce/woocommerce/issues/36413) is merged in Woo core.

As of now, third-party plugins can declare compatibility with the Cart & Checkout Blocks. Declaring (in)compatibility is simple:

- Declare WC tested up to in the main plugin file(s)
- Add the correct code snippet to the main plugin file(s)

The code snippet used for declaring (in)compatibility with the Cart & Checkout Blocks:

```PHP

add_action( 'before_woocommerce_init', function() {
	if ( class_exists( \Automattic\WooCommerce\Utilities\FeaturesUtil::class ) ) {
		\Automattic\WooCommerce\Utilities\FeaturesUtil::declare_compatibility( 'cart_checkout_blocks', __FILE__, true );
	}
} );

```

### Requested feature

We'll need to detect the installed or enabled plugins within WooCommerce Blocks that declare their (in)compatibility with the Cart & Checkout Blocks (i.e., using the code snippet above). Then decide what to do with this information. Here are some propositions:

- Display the plugins declaring their incompatibility in a notice on the Cart & Checkout editor pages.

<img width="287" alt="image" src="https://user-images.githubusercontent.com/14235870/225001461-ef416a6e-3808-434a-9f0f-7e67c79deec9.png">

- Display a warning under the plugin in question on the plugins page

<img width="750" alt="image" src="https://user-images.githubusercontent.com/14235870/225001922-f3bfd565-49a9-43bb-bd5d-35cd4fbd7f8f.png">

- Collect data from store owners who have installed plugins that declare (in)compatibility with the C&C Blocks. We can see how many people are using C&C Bocks as well as some incompatible extensions. (see pca54o-4NP-p2#tracks)

**Comment by tarhi-saad** [comment]
There is an ongoing RFC: pdFofs-Ot-p2

**Comment by tarhi-saad** [comment]
We merged a PR: #10877, that tackles this issue! We chose to display the plugins declaring their incompatibility in a notice on the Cart & Checkout editor pages.
